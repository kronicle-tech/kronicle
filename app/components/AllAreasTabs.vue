<template>
  <div class="my-3">
    <b-button to="/all-areas">Areas</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
