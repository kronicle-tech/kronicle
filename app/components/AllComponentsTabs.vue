<template>
  <div class="my-3">
    <b-button class="my-1" :to="{ name: 'all-components' }">
      Components
    </b-button>
    <b-button
      :to="{ name: 'all-components-tests', query: { testOutcome: 'fail' } }"
    >
      Tests
    </b-button>
    <b-button
      class="my-1"
      :to="{ name: 'all-components-cross-functional-requirements' }"
    >
      Cross Functional Requirements
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-tech-debts' }">
      Tech Debts
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-openapi-specs' }">
      OpenAPI Specs
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-graphql-schemas' }">
      GraphQL Schemas
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-key-software' }">
      Key Software
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-repos' }">
      Repos
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-errors' }">
      Errors
    </b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
