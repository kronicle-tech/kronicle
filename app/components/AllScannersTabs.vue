<template>
  <div class="my-3">
    <b-button :to="{ name: 'all-scanners' }">Scanners</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
