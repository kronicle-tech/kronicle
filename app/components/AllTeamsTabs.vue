<template>
  <div class="my-3">
    <b-button to="/all-teams">Teams</b-button>
    <b-button to="/all-teams/components">Components</b-button>
    <b-button to="/all-teams/tech-debts">Tech Debts</b-button>
    <b-button to="/all-teams/tests">Tests</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
