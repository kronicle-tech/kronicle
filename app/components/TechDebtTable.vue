<template>
  <table v-if="techDebts && techDebts.length > 0" class="table table-dark">
    <thead>
      <tr>
        <th class="component">Component</th>
        <th class="priority">Priority</th>
        <th class="teams">Teams</th>
        <th class="description">Description</th>
        <th class="notes">Notes</th>
        <th class="links">Links</th>
      </tr>
    </thead>
    <tbody>
      <tr v-for="(techDebt, techDebtIndex) in techDebts" :key="techDebtIndex">
        <td class="component table-primary">
          <ComponentName :component="techDebt.component" />
        </td>
        <td class="priority">
          <b-badge :variant="techDebt.priorityVariant">
            {{ techDebt.priorityText }}
          </b-badge>
        </td>
        <td class="teams">
          <ComponentTeams :component-teams="techDebt.component.teams" />
        </td>
        <td class="description">
          <Markdown :markdown="techDebt.description" />
        </td>
        <td class="notes">
          <Markdown :markdown="techDebt.notes" />
        </td>
        <td class="links">
          <Links :links="techDebt.links" />
        </td>
      </tr>
    </tbody>
  </table>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { BBadge } from 'bootstrap-vue'
import { Component, TechDebt } from '~/types/component-catalog-service'
import { getPriorityText, getPriorityVariant } from '~/src/priorityHelper'
import ComponentName from '~/components/ComponentName.vue'
import Links from '~/components/Links.vue'
import Markdown from '~/components/Markdown.vue'
import ComponentTeams from '~/components/ComponentTeams.vue'
import { compareObjectsWithPriorities } from '~/src/objectWithPriorityComparator'

interface TechDebtWithComponent extends TechDebt {
  component: Component
  priorityVariant: string
  priorityText: string
}

export default Vue.extend({
  components: {
    'b-badge': BBadge,
    ComponentName,
    ComponentTeams,
    Links,
    Markdown,
  },
  props: {
    components: {
      type: Array as PropType<Component[]>,
      required: true,
    },
  },
  computed: {
    techDebts(): TechDebtWithComponent[] {
      const that = this
      return that.components
        .flatMap((component) => {
          return (component.techDebts ?? []).map((techDebt) => {
            return {
              ...techDebt,
              component,
              priorityVariant: getPriorityVariant(techDebt.priority),
              priorityText: getPriorityText(techDebt.priority),
            }
          })
        })
        .sort(compareObjectsWithPriorities)
    },
  },
})
</script>
