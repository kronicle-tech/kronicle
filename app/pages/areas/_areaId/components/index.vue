<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <h1 class="text-info my-3">{{ area.name }} Area - Components</h1>

          <AreaTabs :area-id="area.id" />
        </b-col>
      </b-row>
    </b-container>

    <ComponentsView :components="area.components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BCol, BContainer, BRow} from "bootstrap-vue";
import { Area } from '~/types/kronicle-service'
import AreaTabs from '~/components/AreaTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AreaTabs,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const area = await fetch(
      `${$config.serviceBaseUrl}/v1/areas/${route.params.areaId}?fields=area(id,name,components(id,name,typeId,description,tags,teams,platformId))`
    )
      .then((res) => res.json())
      .then((json) => json.area as Area)

    store.commit('componentFilters/initialize', {
      components: area.components,
      route,
    })

    return {
      area,
    }
  },
  data() {
    return {
      area: {} as Area,
    }
  },
  head(): MetaInfo {
    return {
      title: `Kronicle - ${this.area.name} Area - Components`,
    }
  },
})
</script>
