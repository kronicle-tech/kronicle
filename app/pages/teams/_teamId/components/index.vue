<template>
  <div class="m-3">
    <h1 class="text-info my-3">{{ team.name }} - Components</h1>

    <TeamTabs :team-id="team.id" />

    <ComponentsView :components="team.components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import { Team } from '~/types/kronicle-service'
import ComponentsView from '~/components/ComponentsView.vue'
import TeamTabs from '~/components/TeamTabs.vue'
import { NuxtError } from '~/src/nuxtError'

export default Vue.extend({
  components: {
    ComponentsView,
    TeamTabs,
  },
  async asyncData({ $config, route, store }) {
    const team = await fetch(
      `${$config.serviceBaseUrl}/v1/teams/${route.params.teamId}?fields=team(id,name,components(id,name,discovered,type,description,tags,teams,platformId))`
    )
      .then((res) => res.json())
      .then((json) => json.team as Team | undefined)

    if (!team) {
      throw new NuxtError('Team not found', 404)
    }

    store.commit('componentFilters/initialize', {
      components: team.components,
      route,
    })

    return {
      team,
    }
  },
  data() {
    return {
      team: {} as Team,
    }
  },
  head(): MetaInfo {
    return {
      title: `Kronicle - ${this.team.name} - Components`,
    }
  },
})
</script>
