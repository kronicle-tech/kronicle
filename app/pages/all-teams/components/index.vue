<template>
  <div class="m-3">
    <AllTeamsTabs />

    <b-card-group v-for="(row, rowIndex) in rows" :key="rowIndex" deck>
      <b-card
        v-for="(item, itemIndex) in row"
        :key="itemIndex"
        bg-variant="dark"
        :header="item.team.name"
        text-variant="white"
        class="my-3"
      >
        <b-card
          v-for="component in item.components"
          :key="component.id"
          no-body
          class="mt-1"
        >
          <b-card-header header-tag="header" class="p-1" role="tab">
            <b-button
              v-b-toggle="`accordion-${rowIndex}-${itemIndex}-${component.id}`"
              block
              variant="info"
              >{{ component.name }}</b-button
            >
          </b-card-header>
          <b-collapse
            :id="`accordion-${rowIndex}-${itemIndex}-${component.id}`"
            visible
            :accordion="`accordion-${rowIndex}-${itemIndex}`"
            role="tabpanel"
          >
            <b-card-body>
              <ComponentPanel :component="component" />
            </b-card-body>
          </b-collapse>
        </b-card>
      </b-card>
    </b-card-group>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import {
  BButton,
  BCard,
  BCardBody,
  BCardGroup,
  BCardHeader,
  BCollapse,
  VBToggle,
} from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component, Team } from '~/types/kronicle-service'
import AllTeamsTabs from '~/components/AllTeamsTabs.vue'
import ComponentPanel from '~/components/ComponentPanel.vue'

interface Item {
  team: Team
  components: Component[]
}

export default Vue.extend({
  components: {
    AllTeamsTabs,
    'b-button': BButton,
    'b-card': BCard,
    'b-card-body': BCardBody,
    'b-card-group': BCardGroup,
    'b-card-header': BCardHeader,
    'b-collapse': BCollapse,
    ComponentPanel,
  },
  directives: {
    'b-toggle': VBToggle,
  },
  async asyncData({ $config }) {
    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id,name)`
    )
      .then((res) => res.json())
      .then((json) => json.teams)

    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,teams,tags,description,notes,responsibilities,techDebts,openApiSpecs,links)`
    )
      .then((res) => res.json())
      .then((json) => json.components)

    return {
      teams,
      components,
    }
  },
  data() {
    return {
      teams: [] as Team[],
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Teams',
    }
  },
  computed: {
    rows() {
      const that = this
      const rows = [] as Array<Array<Item>>
      const teamCount = this.teams.length
      let startIndex = 0
      while (startIndex < teamCount - 1) {
        const endIndex = Math.min(startIndex + 3, teamCount)
        const row = that.teams.slice(startIndex, endIndex).map(
          (team) =>
            ({
              team,
              components: getComponents(team),
            } as Item)
        )
        rows.push(row)
        startIndex = endIndex
      }
      return rows

      function getComponents(team: Team) {
        return that.components.filter((component) => {
          return component.teams.some((componentTeam) => {
            return (
              componentTeam.type !== 'previous' &&
              componentTeam.teamId === team.id
            )
          })
        })
      }
    },
  },
})
</script>
