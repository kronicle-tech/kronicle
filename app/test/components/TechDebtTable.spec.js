import { mount } from '@vue/test-utils'
import TechDebtTable from '@/components/TechDebtTable.vue'
import { createComponentWithTechDebts } from '~/test/testDataUtils'
import { expectTextsInTableRows } from '~/test/components/tableUtils'

describe('TechDebtTable', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(TechDebtTable, { propsData })
  }

  beforeEach(() => {
    propsData = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when components prop is set to an empty array', () => {
    beforeEach(() => {
      propsData.components = []
    })

    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
    })
  })

  describe('when components is set to multiple components with multiple tech debts', () => {
    beforeEach(() => {
      propsData.components = [
        createComponentWithTechDebts({ componentNumber: 1 }),
        createComponentWithTechDebts({ componentNumber: 2 }),
        createComponentWithTechDebts({ componentNumber: 3 }),
      ]
    })

    test('renders the tech debts', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
      expectTextsInTableRows(wrapper, 'td.description p', [
        'Test Tech Debt Description 1 2',
        'Test Tech Debt Description 2 2',
        'Test Tech Debt Description 3 2',
        'Test Tech Debt Description 1 1',
        'Test Tech Debt Description 2 1',
        'Test Tech Debt Description 3 1',
      ])
    })
  })
})
