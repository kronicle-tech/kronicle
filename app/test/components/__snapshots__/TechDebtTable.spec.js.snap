// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechDebtTable when components is set to multiple components with multiple tech debts renders the tech debts 1`] = `
"<table class=\\"table table-dark\\">
  <thead>
    <tr>
      <th class=\\"component\\">Component</th>
      <th class=\\"priority\\">Priority</th>
      <th class=\\"teams\\">Teams</th>
      <th class=\\"description\\">Description</th>
      <th class=\\"notes\\">Notes</th>
      <th class=\\"links\\">Links</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class=\\"component table-primary\\"><a href=\\"/components/test-component-id-1\\">Test Component Name 1</a></td>
      <td class=\\"priority\\"><span class=\\"badge badge-warning\\">
          High
        </span></td>
      <td class=\\"teams\\">
        <ul>
          <li><a href=\\"/teams/test-team-id-1-1\\" class=\\"team-name\\">test-team-id-1-1</a>
            <!---->
            <!---->
          </li>
          <li><a href=\\"/teams/test-team-id-1-2\\" class=\\"team-name\\">test-team-id-1-2</a>
            <!---->
            <!---->
          </li>
        </ul>
      </td>
      <td class=\\"description\\">
        <div>
          <p>Test Tech Debt Description 1 2</p>
        </div>
      </td>
      <td class=\\"notes\\">
        <div>
          <p>Test Tech Debt Notes 1 2</p>
        </div>
      </td>
      <td class=\\"links\\">
        <ul>
          <li><a href=\\"https://example.com/test-tech-debt-link-1-2-1\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-1-2-1</a> <br>
            <div>
              <p>Test Tech Debt Link Description 1 2 1</p>
            </div>
          </li>
          <li><a href=\\"https://example.com/test-tech-debt-link-1-2-2\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-1-2-2</a> <br>
            <div>
              <p>Test Tech Debt Link Description 1 2 2</p>
            </div>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td class=\\"component table-primary\\"><a href=\\"/components/test-component-id-2\\">Test Component Name 2</a></td>
      <td class=\\"priority\\"><span class=\\"badge badge-warning\\">
          High
        </span></td>
      <td class=\\"teams\\">
        <ul>
          <li><a href=\\"/teams/test-team-id-2-1\\" class=\\"team-name\\">test-team-id-2-1</a>
            <!---->
            <!---->
          </li>
          <li><a href=\\"/teams/test-team-id-2-2\\" class=\\"team-name\\">test-team-id-2-2</a>
            <!---->
            <!---->
          </li>
        </ul>
      </td>
      <td class=\\"description\\">
        <div>
          <p>Test Tech Debt Description 2 2</p>
        </div>
      </td>
      <td class=\\"notes\\">
        <div>
          <p>Test Tech Debt Notes 2 2</p>
        </div>
      </td>
      <td class=\\"links\\">
        <ul>
          <li><a href=\\"https://example.com/test-tech-debt-link-2-2-1\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-2-2-1</a> <br>
            <div>
              <p>Test Tech Debt Link Description 2 2 1</p>
            </div>
          </li>
          <li><a href=\\"https://example.com/test-tech-debt-link-2-2-2\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-2-2-2</a> <br>
            <div>
              <p>Test Tech Debt Link Description 2 2 2</p>
            </div>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td class=\\"component table-primary\\"><a href=\\"/components/test-component-id-3\\">Test Component Name 3</a></td>
      <td class=\\"priority\\"><span class=\\"badge badge-warning\\">
          High
        </span></td>
      <td class=\\"teams\\">
        <ul>
          <li><a href=\\"/teams/test-team-id-3-1\\" class=\\"team-name\\">test-team-id-3-1</a>
            <!---->
            <!---->
          </li>
          <li><a href=\\"/teams/test-team-id-3-2\\" class=\\"team-name\\">test-team-id-3-2</a>
            <!---->
            <!---->
          </li>
        </ul>
      </td>
      <td class=\\"description\\">
        <div>
          <p>Test Tech Debt Description 3 2</p>
        </div>
      </td>
      <td class=\\"notes\\">
        <div>
          <p>Test Tech Debt Notes 3 2</p>
        </div>
      </td>
      <td class=\\"links\\">
        <ul>
          <li><a href=\\"https://example.com/test-tech-debt-link-3-2-1\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-3-2-1</a> <br>
            <div>
              <p>Test Tech Debt Link Description 3 2 1</p>
            </div>
          </li>
          <li><a href=\\"https://example.com/test-tech-debt-link-3-2-2\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-3-2-2</a> <br>
            <div>
              <p>Test Tech Debt Link Description 3 2 2</p>
            </div>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td class=\\"component table-primary\\"><a href=\\"/components/test-component-id-1\\">Test Component Name 1</a></td>
      <td class=\\"priority\\"><span class=\\"badge badge-info\\">
          Medium
        </span></td>
      <td class=\\"teams\\">
        <ul>
          <li><a href=\\"/teams/test-team-id-1-1\\" class=\\"team-name\\">test-team-id-1-1</a>
            <!---->
            <!---->
          </li>
          <li><a href=\\"/teams/test-team-id-1-2\\" class=\\"team-name\\">test-team-id-1-2</a>
            <!---->
            <!---->
          </li>
        </ul>
      </td>
      <td class=\\"description\\">
        <div>
          <p>Test Tech Debt Description 1 1</p>
        </div>
      </td>
      <td class=\\"notes\\">
        <div>
          <p>Test Tech Debt Notes 1 1</p>
        </div>
      </td>
      <td class=\\"links\\">
        <ul>
          <li><a href=\\"https://example.com/test-tech-debt-link-1-1-1\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-1-1-1</a> <br>
            <div>
              <p>Test Tech Debt Link Description 1 1 1</p>
            </div>
          </li>
          <li><a href=\\"https://example.com/test-tech-debt-link-1-1-2\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-1-1-2</a> <br>
            <div>
              <p>Test Tech Debt Link Description 1 1 2</p>
            </div>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td class=\\"component table-primary\\"><a href=\\"/components/test-component-id-2\\">Test Component Name 2</a></td>
      <td class=\\"priority\\"><span class=\\"badge badge-info\\">
          Medium
        </span></td>
      <td class=\\"teams\\">
        <ul>
          <li><a href=\\"/teams/test-team-id-2-1\\" class=\\"team-name\\">test-team-id-2-1</a>
            <!---->
            <!---->
          </li>
          <li><a href=\\"/teams/test-team-id-2-2\\" class=\\"team-name\\">test-team-id-2-2</a>
            <!---->
            <!---->
          </li>
        </ul>
      </td>
      <td class=\\"description\\">
        <div>
          <p>Test Tech Debt Description 2 1</p>
        </div>
      </td>
      <td class=\\"notes\\">
        <div>
          <p>Test Tech Debt Notes 2 1</p>
        </div>
      </td>
      <td class=\\"links\\">
        <ul>
          <li><a href=\\"https://example.com/test-tech-debt-link-2-1-1\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-2-1-1</a> <br>
            <div>
              <p>Test Tech Debt Link Description 2 1 1</p>
            </div>
          </li>
          <li><a href=\\"https://example.com/test-tech-debt-link-2-1-2\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-2-1-2</a> <br>
            <div>
              <p>Test Tech Debt Link Description 2 1 2</p>
            </div>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td class=\\"component table-primary\\"><a href=\\"/components/test-component-id-3\\">Test Component Name 3</a></td>
      <td class=\\"priority\\"><span class=\\"badge badge-info\\">
          Medium
        </span></td>
      <td class=\\"teams\\">
        <ul>
          <li><a href=\\"/teams/test-team-id-3-1\\" class=\\"team-name\\">test-team-id-3-1</a>
            <!---->
            <!---->
          </li>
          <li><a href=\\"/teams/test-team-id-3-2\\" class=\\"team-name\\">test-team-id-3-2</a>
            <!---->
            <!---->
          </li>
        </ul>
      </td>
      <td class=\\"description\\">
        <div>
          <p>Test Tech Debt Description 3 1</p>
        </div>
      </td>
      <td class=\\"notes\\">
        <div>
          <p>Test Tech Debt Notes 3 1</p>
        </div>
      </td>
      <td class=\\"links\\">
        <ul>
          <li><a href=\\"https://example.com/test-tech-debt-link-3-1-1\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-3-1-1</a> <br>
            <div>
              <p>Test Tech Debt Link Description 3 1 1</p>
            </div>
          </li>
          <li><a href=\\"https://example.com/test-tech-debt-link-3-1-2\\" target=\\"_blank\\">https://example.com/test-tech-debt-link-3-1-2</a> <br>
            <div>
              <p>Test Tech Debt Link Description 3 1 2</p>
            </div>
          </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>"
`;
